tesseract_process_managers/include/tesseract_process_managers/taskflow_generators/graph_taskflow.h

```diff
@@ -96,23 +62,36 @@ class GraphTaskflow : public TaskflowGenerator
   /**
    * @brief Add a node to the taskflow graph along with setting the process type.
    * @param process The process generator assigned to the node
-   * @param process_type The process type assigned to the node
+   * @param is_conditional Flag indicating if this process is conditional
    * @return The node ID which should be used with adding edges
    */
-  int addNode(TaskGenerator::UPtr process, NodeType process_type);
+  int addNode(TaskGenerator::UPtr process, bool is_conditional = false);
 
   /**
-   * @brief Add an edge to the taskflow graph
-   * @param src The source node ID
-   * @param src_channel The source channel (ON_SUCCESS, ON_FAILURE)
-   * @param dest The destination node ID (This is only required for destination channels PROCESS_NODE) othewise pass a
-   * -1
-   * @param dest_channel The destination channel to connect with the source channel (PROCESS_NODE, DONE_CALLBACK,
-   * ERROR_CALLBACK).
+   * @brief Adds directed edges from a source node to desintation nodes in the taskflow graph
+   * @details If source is a non-conditional task, it is only relevant to provide one destination as the output of a
+   * non-conditional tf::Task is void. If source is a conditional task, the order of the destinations should correspond
+   * to the integer output of the conditional tf::Task. For example,if the output of the conditional tf::Task is 0, the
+   * taskflow would transition to the node identified by the ID at index 0 in the destinations input. A leaf node (i.e.
+   * node without defined edges) is always connected to the done callback. If the leaf node is a conditional node, it is
+   * connected to the error task at index 0 and the done task at index 1
+   * @param from The ID of the source node of the edge
+   * @param dest A list of IDs of the destination nodes of the edges.
    */
-  void addEdge(int src, SourceChannel src_channel, int dest, DestinationChannel dest_channel);
+  void addEdges(int source, std::vector<int> destinations);
 
 private:
+  /** @brief Helper struct for holding relevant information about the nodes */
+  struct Node
+  {
+    Node(TaskGenerator::UPtr process_, const bool is_conditional_);
```

Comment:
Clang-Tidy remove const before bool. Should the parameters have an underscore at the end?

Reply:
I added the underscores to differentiate the arguments from the struct members [in the initializer list](https://github.com/marip8/tesseract_planning/blob/38afc2aba11cf8214bf764fc73ef77ea8eff60c2/tesseract_process_managers/src/taskflow_generators/graph_taskflow.cpp#L41)